src/providers/packet/mod.rs

```diff
@@ -117,83 +118,103 @@ fn get_dns_servers() -> Result<Vec<IpAddr>> {
 
 fn parse_network(netinfo: &PacketNetworkInfo) -> Result<(Vec<Interface>,Vec<Device>)> {
     let mut interfaces = Vec::new();
+    let mut bonds = Vec::new();
     for i in netinfo.interfaces.clone() {
         let mac = MacAddr::from_str(&i.mac)
             .map_err(|err| Error::from(format!("{:?}", err)))
             .chain_err(|| format!("failed to parse mac address: '{}'", i.mac))?;
         interfaces.push(Interface {
             mac_address: Some(mac),
-            bond: Some("bond0".to_owned()),
+            bond: i.bond.clone(),
             name: None,
             priority: None,
             nameservers: Vec::new(),
             ip_addresses: Vec::new(),
             routes: Vec::new(),
         });
+
+        // if there is a bond key, make sure we have a bond device for it
+        if let Some(ref bond_name) = i.bond {
+            let bond = Interface {
+                name: Some(bond_name.clone()),
+                priority: Some(5),
+                nameservers: get_dns_servers()?,
+                mac_address: Some(mac),
+                bond: None,
+                ip_addresses: Vec::new(),
+                routes: Vec::new(),
+            };
+            if !bonds.contains(&bond) {
```

Comment:
Won't this always return false because of `mac_address`?

Reply:
it would do that. I didn't notice because it created another file of the same name with almost the same contents that overwrote the other one, so it still emitted the right number of bond devices